Jenkinsfile

```diff
@@ -0,0 +1,153 @@
+import groovy.json.JsonOutput
+
+/** Tox environment */
+def environment = 'py27'
+
+/** Map of desired capabilities */
+def capabilities = [
+  browserName: 'Firefox',
+  version: '47.0',
+  platform: 'Windows 7'
+]
+
+/** Write capabilities to JSON file
+ *
+ * @param desiredCapabilities capabilities to include in the file
+*/
+def writeCapabilities(desiredCapabilities) {
+    def defaultCapabilities = [
+        build: env.BUILD_TAG,
+        public: 'public restricted'
+    ]
+    def capabilities = defaultCapabilities.clone()
+    capabilities.putAll(desiredCapabilities)
+    def json = JsonOutput.toJson([capabilities: capabilities])
+    writeFile file: 'capabilities.json', text: json
+}
+
+/** Run Tox
+ *
+ * @param environment test environment to run
+*/
+def runTox(environment) {
+  def processes = env.PYTEST_PROCESSES ?: 'auto'
+  try {
+    wrap([$class: 'AnsiColorBuildWrapper']) {
+      withCredentials([[
+        $class: 'StringBinding',
+        credentialsId: 'SAUCELABS_API_KEY',
+        variable: 'SAUCELABS_API_KEY']]) {
+        withEnv(["PYTEST_ADDOPTS=${PYTEST_ADDOPTS} " +
+          "-n=${processes} " +
+          "--driver=SauceLabs " +
+          "--variables=capabilities.json " +
+          "--color=yes"]) {
+          sh "tox -e ${environment}"
+        }
+      }
+    }
+  } catch(err) {
+    currentBuild.result = 'FAILURE'
+    throw err
+  } finally {
+    dir('results') {
+      stash environment
+    }
+  }
+}
+
+/** Send a notice to #fxtest-alerts on irc.mozilla.org with the build result
+ *
+ * @param result outcome of build
+*/
+def ircNotification(result) {
+  def nick = "fxtest${BUILD_NUMBER}"
+  def channel = '#fx-test-alerts'
+  result = result.toUpperCase()
+  def message = "Project ${JOB_NAME} build #${BUILD_NUMBER}: ${result}: ${BUILD_URL}"
+  node {
+    sh """
+        (
+        echo NICK ${nick}
+        echo USER ${nick} 8 * : ${nick}
+        sleep 5
+        echo "JOIN ${channel}"
+        echo "NOTICE ${channel} :${message}"
+        echo QUIT
+        ) | openssl s_client -connect irc.mozilla.org:6697
+    """
+  }
+}
+
+stage('Checkout') {
+  node {
+    timestamps {
+      deleteDir()
+      checkout scm
+      stash 'workspace'
```

Comment:
The failure on staging is suggesting that the `.git` directory is not being preserved (and therefore it can't determine the latest tag to use for the version name). We could either hardcode a version number, or we could try to preserve the `.git` directory.

Try changing this line to:

```
stash 'workspace', useDefaultExcludes: false
```

If that doesn't work, we add the `SETUPTOOLS_SCM_PRETEND_VERSION` environment variable with a dummy version number where we execute Tox.